spec/rubocop/cop/layout/empty_line_after_guard_clause_spec.rb

```diff
@@ -560,4 +560,15 @@ def foo
       end
     RUBY
   end
+
+  it 'registers an offense but does not correct when the clause is ended with a semicolon' do
+    expect_offense(<<~RUBY)
+      def foo(item)
+        return unless item.positive?; item * 2
+        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Add empty line after guard clause.
```

Comment:
I think this case is acceptable because it is a false positive. So the cop can respect the will of programmers who want to use semicolons.

Reply:
@koic I'm not sure I follow, that offense isn't new, this PR just doesn't try to autocorrect the offense.

Are you saying this should not be an offense?

Reply:
Yes. I think `expect_no_offences` should be used in this test and pass for cop role’s design.

Reply:
@koic updated, it'll now not consider guard clauses when there's another statement on the same line. I added some additional tests as well.

Reply:
Great! Thanks!